api/views.py

```diff
@@ -1,41 +1,50 @@
+"""
+# Collection Views
+## Defines
+- CollectionViewSet
+- ItemViewSet
+- ColletionMembershipViweSet
+- DocumentviewSet
+"""
+
+
+# Imports
+# #############################################################################
+
+
+from django.http import HttpResponse
 from django.db.models import Q
 from django.core.exceptions import ObjectDoesNotExist
 from rest_framework import generics
+from rest_framework.viewsets import ModelViewSet
 from rest_framework_json_api import pagination
 from rest_framework import exceptions as drf_exceptions
 from rest_framework import permissions as drf_permissions
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
 from rest_framework.reverse import reverse
-from api.serializers import CollectionSerializer, CollectionSearchSerializer, \
-    ItemSerializer, ItemSearchSerializer, \
-    UserSerializer, UserSearchSerializer
-from api.models import Collection, Item, User
-from api.permissions import CanEditCollection, CanEditItem
-
 from drf_haystack.viewsets import HaystackViewSet
+from guardian.shortcuts import assign_perm
 
+from api.models import (
+    Collection,
+    Item,
+    User
+)
+from api.serializers import (
+    CollectionSerializer,
+    ItemSerializer,
+    UserSerializer,
+    CollectionSearchSerializer,
+    ItemSearchSerializer,
+    UserSearchSerializer
+)
 
-class CollectionSearchView(HaystackViewSet):
-    index_models = [Collection]
-    serializer_class = CollectionSearchSerializer
 
-
-class ItemSearchView(HaystackViewSet):
-    index_models = [Item]
-    serializer_class = ItemSearchSerializer
+# Views
+# #############################################################################
 
 
-class UserSearchView(HaystackViewSet):
-    index_models = [User]
-    serializer_class = UserSearchSerializer
-
-
-class LargeResultsSetPagination(pagination.PageNumberPagination):
```

Comment:
Why did you delete this? Removing it would break how pagination against the items endpoint is handled on the frontend.